tests/integration/project/cmd_project_test.go

```diff
@@ -49,6 +49,17 @@ var _ = Describe("odo project command tests", func() {
 		})
 	})
 
+	Context("odo machine readable output on empty project", func() {
+		It("should be able to list current project", func() {
+			projectListJSON := helper.CmdShouldPass("odo", "project", "list", "-o", "json")
+			listOutputJSON, err := helper.Unindented(projectListJSON)
+			Expect(err).Should(BeNil())
+			partOfProjectListJSON, err := helper.Unindented(`{"kind":"Project","apiVersion":"odo.dev/v1alpha1","metadata":{"name":"` + project + `","namespace":"` + project + `","creationTimestamp":null},"spec":{},"status":{"active":true}}`)
+			Expect(err).Should(BeNil())
+			Expect(listOutputJSON).To(ContainSubstring(partOfProjectListJSON))
```

Comment:
It wold be better to use `MatchJSON` instead of `ContainSubstring`

Reply:
@kadel yes, this a better option but it won't work the test scenario we are verifying. We are not matching the whole json here, some part of the json is being matched with the original output because ```projectListJSON := helper.CmdShouldPass("odo", "project", "list", "-o", "json")``` returns a huge output in CI and in locally the size depends on how many projects are there in that cluster when you ran the test script.

Reply:
Agreed, match json might not work as we are comparing a substring not the whole json. But then my question is why not? Why not compare the whole json output using MatchJSON seems more robust to me?

Reply:
> But then my question is why not? Why not compare the whole json output using MatchJSON seems more robust to me?

@girishramnani json length is not the only concern, the major concern is the json length is dynamic. For example namespace ```myproject``` and ```ci-operator-hub-project``` are two default project we get when we run our CI test. So may be down the line we can add more namespace through the configure script. This can lead to change in json length. 

The worst part we will face when we run the test script locally. For example you have a cluster and you running your test as ```kebeadmin``` then ```helper.CmdShouldPass("odo", "project", "list", "-o", "json")``` will list out almost 30 project details. It is weird to add a huge json list to match in test script.

Reply:
So to over come this i am just matching the key portion of the json list